python/ray/dag/tests/experimental/test_torch_tensor_dag.py

```diff
@@ -268,6 +283,54 @@ def test_torch_tensor_nccl(ray_start_regular):
     # ray.get(receiver.ping.remote())
 
 
+@pytest.mark.parametrize(
+    "ray_start_regular, overlap_gpu_communication",
+    [({"num_cpus": 4}, False), ({"num_cpus": 4}, True)],
+    indirect=["ray_start_regular"],
+)
+def test_torch_tensor_nccl_overlap(ray_start_regular, overlap_gpu_communication):
```

Comment:
This seems fine for correctness but it would be better if we can also test that overlapping is actually occurring as expected.

Can we mock out the `wait` and/or cuda calls to make sure that they get called at the right times?

Reply:
I tried to use a mock but have seen issues the mock does not take effect when passing to the actor. Tried several ways but still without luck. I will work on this as a follow up: https://github.com/ray-project/ray/issues/48470

The time saving (see description) when turning on overlap should be a good indicator that the overlap takes effect.